### PR TITLE
Skr grpc tests

### PR DESCRIPTION
**Pull request by hgarvison**
Adding GRPC tests for the updated SKR image

Passing Pipeline using Confidential Sidecar Branch "feature-consolidate-containers" https://github.com/microsoft/confidential-aci-examples/actions/runs/6836143084

### CODE REVIEW

File: examples/key_release/Dockerfile.primary [hunk 1]
```diff
@@ -1,5 +1,14 @@
-FROM python:latest
+FROM alpine:3.18.4
```
Comment (SethHollandsworth) [inline examples/key_release/Dockerfile.primary:1]: just a note we'll need to update this in future to be either ubuntu or mariner

**Review by SethHollandsworth — DISMISSED** [review]
lgtm

**Comment by SethHollandsworth** [comment]
[most up to date pipeline run](https://github.com/microsoft/confidential-aci-examples/actions/runs/6852967191)

**Comment by KenGordon** [comment]
@DomAyre Is there some issue with python versions causing the failure like:
/home/runner/.local/lib/python3.10/site-packages/msal_extensions/cache_lock.py:7: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
<frozen importlib._bootstrap>:671: ImportWarning: DynamicImporter.exec_module() not found; falling back to load_module()
ERROR: argument --username/-u: expected one argumen

File: examples/key_release/payload.py [hunk 2]
```diff
@@ -2,9 +2,29 @@
 # Licensed under the MIT License.
 
 from http.server import BaseHTTPRequestHandler, HTTPServer
+import os
+import subprocess
 
 import requests
 
+def grpc_ready_test():
```
Comment (KenGordon) [inline examples/key_release/payload.py:10]: I wonder if having two distinct example, one using the REST endpoint, one using the GRPC one, would be easier for a user to understand.

Comment (SethHollandsworth) [reply examples/key_release/payload.py:10]: I believe that's what this [set of tests](https://github.com/microsoft/confidential-aci-examples/blob/87659ba3793904d897b734a4a78f1de5091a8067/examples/key_release/test.py#L20) is doing

**Review by DomAyre — APPROVED** [review]
I re-ran the failing local workflow, I think this was a temporary issue with dependencies not with the actual code

**Pull request merged**